### PR TITLE
Add Reader Test Streams for testing variations of cold start post recommendations

### PR DESCRIPTION
**Pull request by gibrown**
First mostly working attempt at this. Still need to finish adjusting the api side.

### CODE REVIEW

File: client/lib/feed-stream-store/index.js [hunk 1]
```diff
@@ -228,7 +242,15 @@ function feedStoreFactory( storeId ) {
 		} );
 	} else if ( storeId === 'recommendations_posts' ) {
 		store = getStoreForRecommendedPosts( storeId );
-	} else if ( storeId === 'coldstart_posts' ) {
+	} else if ( storeId === 'cold_posts' ) {
```
Comment (blowery) [inline client/lib/feed-stream-store/index.js]: maybe just a `else if ( startsWith( storeId, 'cold_posts' ) )` to kill some of the duplication here?
where you also have `import { startsWith } from 'lodash'` in the import section

File: client/reader/recommendations/index.js [hunk 2]
```diff
@@ -38,5 +38,33 @@ export default function() {
 			readerController.removePost,
 			readerController.sidebar,
 			controller.recommendedPosts );
+		page( '/recommendations/cold1w',
```
Comment (blowery) [inline client/reader/recommendations/index.js]: hrm. maybe we could bundle this set up and use page.apply to call it for each path variant? like 
```
forEach( [ path, path, path ], 
    ( path ) => { 
        page.apply( page, [ 
            path, 
            readerController.preloadReaderBundle,
            readerController.loadSubscriptions,
            readerController.updateLastRoute,
            readerController.removePost,
            readerController.sidebar,
            controller.recommendedPosts
] ) } )
```

**Comment by gibrown** [comment]
@blowery one more look when you get a chance.

File: client/lib/feed-stream-store/index.js [hunk 3]
```diff
@@ -1,4 +1,9 @@
 /**
+ * External dependencies
+ */
+import startsWith from 'lodash/startsWith';
```
Comment (blowery) [inline client/lib/feed-stream-store/index.js]: we prefer the tree-shaking style: `import { startsWith } from 'lodash';`

File: client/reader/recommendations/controller.js [hunk 4]
```diff
@@ -47,17 +47,40 @@ export default {
...
-
-		const RecommendedPostsStore = ( '/recommendations/posts' === basePath )
-			? feedStreamFactory( 'recommendations_posts' )
-			: feedStreamFactory( 'coldstart_posts' );
-
-		const mcKey = ( '/recommendations/posts' === basePath )
-			? 'recommendations_posts'
-			: 'coldstart_posts';
+		let fullAnalyticsPageTitle = '';
+		let RecommendedPostsStore = null;feedStreamFactory( 'recommendations_posts' )
```
Comment (blowery) [inline client/reader/recommendations/controller.js]: last bit of this line intentional?

**Comment by blowery** [comment]
looks good after a couple little things get addressed. feel free to ship it once they're taken care of.

**Pull request merged**